src/python/pants/backend/project_info/rules/dependencies.py

```diff
@@ -20,10 +27,16 @@ def register_options(cls, register):
     super().register_options(register)
     register(
       '--transitive',
-      default=True,
+      default=False,
       type=bool,
       help='Run dependencies against transitive dependencies of targets specified on the command line.',
     )
+    # TODO(#8762): Wire this up. Currently we only support `internal`.
+    register(
+      '--type', type=DependencyType, default=DependencyType.INTERNAL,
```

Comment:
Very open to suggestions for a better name. I couldn't think of anything more natural and generally like this, except that `type` is an overloaded word.